test-app/tests/integration/components/toucan-form/toucan-form-test.gts

```diff
@@ -55,6 +56,79 @@ module('Integration | Component | ToucanForm', function (hooks) {
     assert.dom('[data-test-field]').exists();
   });
 
+  test('it yields `submit` from ember-headless-form', async function (assert) {
+    interface FormData {
+      name?: string;
+      email?: string;
+    }
+
+    assert.expect(3);
+
+    const data: FormData = {};
+
+    const handleSubmit = (data: FormData) => {
+      assert.deepEqual(
+        data,
+        { name: 'text', email: 'a@b.com' },
+        'Expected form data to match'
+      );
+      assert.step('onSubmit');
+    };
+
+    await render(<template>
+      <ToucanForm @data={{data}} @onSubmit={{handleSubmit}} as |form|>
+        <form.Input @label="Input" @name="name" data-name />
+        <form.Input @label="Email" @name="email" data-email />
+
+        <button
+          type="button"
+          data-test-submit
+          {{on "click" form.submit}}
+        >Submit</button>
+      </ToucanForm>
+    </template>);
+
+    await fillIn('[data-name]', 'text');
+    await fillIn('[data-email]', 'a@b.com');
+
+    await click('[data-test-submit]');
+
+    assert.verifySteps(['onSubmit']);
+  });
+
+  test('it yields `reset` from ember-headless-form', async function (assert) {
+    interface FormData {
+      name?: string;
+      email?: string;
+    }
+
+    const data: FormData = {
+      name: 'Simon',
+      email: 'a@b.com',
+    };
+
+    await render(<template>
+      <ToucanForm @data={{data}} as |form|>
+        <form.Input @label="Input" @name="name" data-name />
+        <form.Input @label="Email" @name="email" data-email />
+
+        <button
+          type="button"
+          data-test-reset
+          {{on "click" form.reset}}
+        >Submit</button>
+      </ToucanForm>
+    </template>);
+
+    await fillIn('[data-name]', 'Nicole');
+    await fillIn('[data-email]', 'x@yz.com');
+
+    await click('[data-test-reset]');
+
+    assert.dom('[data-name]').hasValue('Simon');
```

Comment:
Oh nice! Such a great reset

Should we also test that it `resets` to empty form data? (not just the form is prefilled with data)

Also, it would be nice to also update the docs with both yielding `submit` and `reset`?

Reply:
> Also, it would be nice to also update the docs with both yielding submit and reset?

Good call - adding some docs in the latest commit!

> Should we also test that it resets to empty form data? (not just the form is prefilled with data)

I outlined this a bit at https://github.com/CrowdStrike/ember-toucan-core/pull/197/files#r1245937002 - I'm a bit hesitant to add too much repetition between our tests and headless-form, personally.